### PR TITLE
quoted map keys

### PR DESCRIPTION
**Pull request by jlgeering**
---

#### Requirements

- [x] Entry in CHANGELOG.md was created
- [x] Link to documentation on https://yaml.org/ is provided in the PR description
- [x] Functionality is covered by newly created tests

### CODE REVIEW

**Comment by mruoss** [comment]
@jlgeering I see my PR template has no effect! 😕

File: lib/ymlr/encoder.ex [hunk 1]
```diff
@@ -105,7 +105,18 @@ defmodule Ymlr.Encoder do
...
-  defp encode_as_io_list(data, level) when is_binary(data) do
+  defp encode_as_io_list(data, level) when is_binary(data), do: encode_binary(data, level)
+
+  defp encode_as_io_list(data, _) when is_number(data), do: "#{data}"
+
+  defp encode_as_io_list(data, _), do: raise(ArgumentError, message: "The given data #{inspect(data)} cannot be converted to YAML.")
+
+  defp encode_map_key(data) when is_atom(data), do: [Atom.to_string(data), ":"]
+  defp encode_map_key(data) when is_binary(data), do: [encode_binary(data, nil), ":"]
+  defp encode_map_key(data) when is_number(data), do: "#{data}:"
```
Comment (mruoss) [inline lib/ymlr/encoder.ex:116]: Now that the number is in its binary representation, should it not go thrpugh encode_binary as well?

Comment (mruoss) [reply lib/ymlr/encoder.ex:116]: No... guess not...

File: lib/ymlr/encoder.ex [hunk 2]
```diff
@@ -105,7 +105,18 @@ defmodule Ymlr.Encoder do
   end
 
   defp encode_as_io_list(data, _) when is_atom(data), do: Atom.to_string(data)
-  defp encode_as_io_list(data, level) when is_binary(data) do
+  defp encode_as_io_list(data, level) when is_binary(data), do: encode_binary(data, level)
+
+  defp encode_as_io_list(data, _) when is_number(data), do: "#{data}"
```
Comment (mruoss) [inline lib/ymlr/encoder.ex:110]: Is it necessary to change it to binary? Wouldn't io to string take care of that latet if you just returned the number?

Comment (jlgeering) [reply lib/ymlr/encoder.ex:110]: encode_binary/2 decides when to wrap something in quotes and deals with multi-line strings (indentation)

**Review by mruoss — APPROVED** [review]
Let's deal with escape chars in a separate PR. => LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+- Wrap map keys in doouble quotes when required ([#94](https://github.com/ufirstgroup/ymlr/issues/94), [#95](https://github.com/ufirstgroup/ymlr/pull/95))
```
Comment (mruoss) [inline CHANGELOG.md:9]: Subtitle is missing. (Added, Changed, ...)

Comment (jlgeering) [reply CHANGELOG.md:11]: Fixed ;-)

**Pull request merged**